### PR TITLE
Documentation comparing Clump and Stitch

### PR DESCRIPTION
**Pull request by fwbrasil**
@williamboxhall This is the initial draft. Do you remember other main diferences?

### CODE REVIEW

File: STITCH.md [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+
+
+optimized for optinal fetches
+	optimized for non-optional fetches
+
+allows filtering
+	doesn't allow filtering
+
+multiple sourcing strategies
+	only positional sourcing
```
Comment (williamboxhall) [inline STITCH.md]: by positional sourcing you mean like 'zip'?

File: STITCH.md [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+
+
+
+optimized for optinal fetches
+	optimized for non-optional fetches
+
+allows filtering
+	doesn't allow filtering
+
+multiple sourcing strategies
```
Comment (williamboxhall) [inline STITCH.md]: I would also add more elegant sourcing interface

File: STITCH.md (hunk 1 above)
Comment (fwbrasil) [inline STITCH.md]: yep

File: STITCH.md [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+	only positional sourcing
+
+caching
+	no caching
+
+simple execution model
+	query ast + execution plan + simplifications
```
Comment (williamboxhall) [inline STITCH.md]: we are not totally sure whether stitch uses AST or not

File: STITCH.md [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+
+can batch nested flatmaps
```
Comment (williamboxhall) [inline STITCH.md]: By the way I still don't have a full understanding of how this works :P, I'm hoping the [unit test](https://github.com/fwbrasil/clump/issues/36) will help me figure it out

File: STITCH.md (hunk 3 above)
Comment (fwbrasil) [inline STITCH.md]: me too, my idea is let clear that these are only suppositions

**Comment by williamboxhall** [comment]
maybe `exists - doesn't exist` :P

**Comment by fwbrasil** [comment]
@williamboxhall The text is ready for review.

File: STITCH.md [hunk 5]
```diff
@@ -0,0 +1,153 @@
...
+The source instance applies the transformations to represent each fetch result as an ```Option``` by extracting the track ids. 
+
+This means that the final clump value is also optional:
+
+```scala
+val clump = tracksSource.get(222)
+val result: Future[Option[Track]] = clump.get
+```
+
+3. Composition
```
Comment (williamboxhall) [inline STITCH.md]: skips from 1 to 3 here

File: STITCH.md [hunk 6]
```diff
@@ -0,0 +1,153 @@
...
+==============
+
+Given the optimization to deal with optional fetches, the ```Clump``` compositions is also different from Stitch. They have semantics similar to relational database's joins, where not found joined elements make the tuple be filtered-out.
```
Comment (williamboxhall) [inline STITCH.md]: I never thought of it like a database join, that's a really awesome way to describe it

**Comment by williamboxhall** [comment]
Looks really great man :+1:

**Pull request merged**